src/backend/distributed/shared_library_init.c

```diff
@@ -866,3 +887,25 @@ NormalizeWorkerListPath(void)
 					PGC_S_OVERRIDE);
 	free(absoluteFileName);
 }
+
+
+static bool
+StatisticsCollectionGucCheckHook(bool *newval, void **extra, GucSource source)
+{
+#if HAVE_LIBCURL
+	return true;
+#else
+
+	/* if libcurl is not installed, only accept false */
+	if (*newval)
+	{
+		ereport(WARNING, (errmsg("cannot set EnableStatisticsCollection to false "
```

Comment:
Firstly: Shouldn't this say _true_? Or maybe _can only set EnableStatisticsCollection to false_?

Secondly: I believe you should be using `GUC_check_errcode(ERRCODE_FEATURE_NOT_SUPPORTED)` here to set the error code and `GUC_check_errdetail` to set the detail and then returning `false` rather than raising an error.

Thirdly: Maybe just go with a detail message of _Citus was compiled without libcurl support_? I believe the message may be auto-set based on you returning false.

Reply:
Done.